File: app/src/main/java/me/toptas/fancyshowcasesample/SecondActivity.java
package me.toptas.fancyshowcasesample;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import me.toptas.fancyshowcase.FancyShowCaseView;

public class SecondActivity extends BaseActivity {

    @BindView(R.id.button1)
    Button mButton;

    @BindView(R.id.toolbar)
    android.support.v7.widget.Toolbar mToolbar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);
        ButterKnife.bind(this);
        setSupportActionBar(mToolbar);

        focusOnButton();
    }

    /**
     * Focus on a view
     */
    @OnClick(R.id.button1)
    public void focus() {
        if (!isFinishing()) {
            focusOnButton();
        }
    }

    private void focusOnButton() {
        new FancyShowCaseView.Builder(SecondActivity.this)
                .focusOn(mButton)
                .title("Focus a view")
                .fitSystemWindows(true)
                .build()
                .show();
    }

    /**
     * Toggle toolbar visibility
     */
    @OnClick(R.id.button2)
    public void toggleToolbar() {
        if (mToolbar.getVisibility() == View.VISIBLE) {
            mToolbar.setVisibility(View.GONE);
        } else {
            mToolbar.setVisibility(View.VISIBLE);
        }
    }


    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return super.onCreateOptionsMenu(menu);
    }


    /**
     * Shows a FancyShowCaseView that focuses to ActionBar items
     *
     * @param item actionbar item to focus
     * @return true
     */
    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        new FancyShowCaseView.Builder(this)
                .focusOn(findViewById(item.getItemId()))
                .title("Focus on Actionbar items")
                .fitSystemWindows(true)
                .build()
                .show();
        return true;
    }
}

File: app/src/main/java/me/toptas/fancyshowcasesample/SecondActivity.kt
package me.toptas.fancyshowcasesample

import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.view.View

import kotlinx.android.synthetic.main.activity_second.*
import me.toptas.fancyshowcase.FancyShowCaseView

class SecondActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)
        setSupportActionBar(toolbar)

        focusOnButton()

        button1.setOnClickListener {
            focusOnButton()
        }

        button2.setOnClickListener {
            if (toolbar.visibility == View.VISIBLE) {
                toolbar.visibility = View.GONE
            } else {
                toolbar.visibility = View.VISIBLE
            }
        }
    }


    private fun focusOnButton() {
        FancyShowCaseView.Builder(this@SecondActivity)
                .focusOn(button1)
                .title("Focus a view")
                .fitSystemWindows(true)
                .build()
                .show()
    }


    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_main, menu)
        return super.onCreateOptionsMenu(menu)
    }


    /**
     * Shows a FancyShowCaseView that focuses to ActionBar items
     *
     * @param item actionbar item to focus
     * @return true
     */
    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        FancyShowCaseView.Builder(this)
                .focusOn(findViewById(item.itemId))
                .title("Focus on Actionbar items")
                .fitSystemWindows(true)
                .build()
                .show()
        return true
    }
}
